src/kernel/capinv.c

```diff
@@ -552,33 +553,48 @@ cap_asnd_op(struct cap_asnd *asnd, struct thread *thd, struct pt_regs *regs,
 }
 
 int
-capinv_int_snd(struct thread *rcv_thd, struct pt_regs *regs)
+cap_hw_asnd(struct cap_asnd *asnd, struct pt_regs *regs)
 {
-	struct comp_info *ci;
-	struct thread *thd, *next;
-	struct tcap *tcap, *rcv_tcap;
+	int restore_curr_thd = 1;
+	if (asnd->h.type != CAP_ASND) return restore_curr_thd;
```

Comment:
Sorry, I was thinking returning immediately in this case will save time. I'll move it back, weird judgements I make I believe. ;)